src/components/skills-quiz/data/hooks.js

```diff
@@ -35,25 +36,43 @@ export const useSelectedSkillsAndJobSkills = ({ getAllSkills, getAllSkillsWithSi
       }
       return skillsFromJob;
     },
-    [skills, interestedJobs, selectedJob, goal, currentJobRole, getAllSkills],
+    [selectedJob, goal, interestedJobs, currentJobRole, getAllSkillsWithSignificanceOrder],
   );
-  // Top 3 Recommended courses are shown based on job-skills only
-  // But on search page show courses based on job-skills and skills selected in skills dropdown as well
-  if (getAllSkills) {
-    return skills ? skills.concat(skillsFromSelectedJob) : skillsFromSelectedJob;
-  }
-  if (getAllSkillsWithSignificanceOrder) {
+
+  const allSkillsWithoutSignificanceOrder = useMemo(() => {
+    if (skills) {
+      return skills.concat(skillsFromSelectedJob);
+    }
+    return skillsFromSelectedJob;
+  }, [skills, skillsFromSelectedJob]);
+
+  const allSkillsWithSignificantOrder = useMemo(() => {
     const allSkills = [];
     skillsFromSelectedJob.forEach((skill) => allSkills.push({
       key: skill.name,
       value: skill.significance,
     }));
-    // eslint-disable-next-line no-unused-expressions
-    skills?.forEach((skill) => allSkills.push({
-      key: skill,
-      value: undefined,
-    }));
+
+    if (skills) {
+      skills.forEach((skill) => {
+        allSkills.push({
+          key: skill,
+          value: undefined,
+        });
+      });
+    }
```

Comment:
could be allSkills.push([...skills.map((skill) => ({ key: skill, value: undefined })])